src/BloomExe/XmlHtmlConverter.cs

```diff
@@ -95,6 +95,9 @@ public static XmlDocument GetXmlDomFromHtml(string content, bool includeXmlDecla
 						// remove blank lines at the end of style blocks
 						newContents = Regex.Replace(newContents, @"\s+<\/style>", "</style>");
 
+						// remove <br> elements immediately preceding </p> close tag (BL-2557)
```

Comment:
This wants a sizable comment about where these come from, how ideally editor.getData() is supposed to be used, why that didn't work, etc.